File: tools/gen_proto.go
package main

import (
	"flag"
	"fmt"
	"io/ioutil"
	"os"
	"os/exec"
	"path/filepath"
)

var protoDir string

func init() {
	flag.StringVar(&protoDir, "i", "./proto", "protocol files dir")
}

func walkFunc(path string, info os.FileInfo, err error) error {
	if info.IsDir() {
		return nil
	}

	fileSuffix := filepath.Ext(info.Name())
	if fileSuffix == ".proto" {
		fmt.Printf("file: %s\n", path)
		cmd := exec.Command("protoc",
			"--gogofaster_out", protoDir,
			"--proto_path", protoDir,
			"--proto_path", filepath.Join(protoDir, "/thirdparty"),
			info.Name())
		out, err := cmd.CombinedOutput()
		if err != nil {
			fmt.Println(string(out), err)
			return err
		}
		fmt.Println(string(out))
	}

	return nil
}

func main() {
	flag.Parse()

	infos, err := ioutil.ReadDir(protoDir)
	for _, info := range infos {
		walkFunc(filepath.Join(protoDir, info.Name()), info, nil)
	}
	//err := filepath.Walk(protoDir, walkFunc)
	if err != nil {
		fmt.Print(err)
	}
}

File: tools/gen_proto.py

import os,sys

PROTO_DIR = "../proto"

def Main():
    global PROTO_DIR
    if len(sys.argv) > 1:
        PROTO_DIR = sys.argv[1]
    # print(PROTO_DIR)

    dirs = os.listdir(PROTO_DIR)
    # print(dirs)

    thirdparty = os.path.join(PROTO_DIR, "thirdparty")

    for file in dirs:
        ext = os.path.splitext(file)[1]
        if ext  == ".proto":
            cmd = "protoc --gogofaster_out={} --proto_path={} --proto_path={} {}".format(PROTO_DIR, PROTO_DIR, thirdparty, file)
            print(cmd)
            os.system(cmd)

Main()